WiserTaskScheduler/WiserTaskScheduler/Modules/ServerMonitors/Services/ServerMonitorsService.cs

```diff
@@ -0,0 +1,433 @@
+
+            int threshold = monitorItem.Threshold;
+
+            //Check which type of server monitor is used.
+            //Cpu is default Value.
+            switch (monitorItem.ServerMonitorType)
+            {
+                case ServerMonitorTypes.Drive:
+                    if (monitorItem.DriveName == "All")
+                    {
+                        await GetAllHardDrivesSpaceAsync(monitorItem, threshold, gclCommunicationsService, configurationServiceName);
+                    }
+                    else
+                    {
+                        await GetHardDriveSpaceAsync(monitorItem, threshold, gclCommunicationsService, configurationServiceName, monitorItem.DriveName);
+                    }
+                    break;
+                case ServerMonitorTypes.Ram:
+                    await GetRAMSpaceAsync(monitorItem, threshold, gclCommunicationsService, configurationServiceName);
+                    break;
+                case ServerMonitorTypes.Cpu:
+                    await GetCpuUsageAsync(monitorItem, threshold, gclCommunicationsService, configurationServiceName);
+                    break;
+                case ServerMonitorTypes.Network:
+                    await GetNetworkUtilization(monitorItem, threshold, gclCommunicationsService, configurationServiceName);
+                    break;
+                default:
+                    break;
+            }
+
+            return new JObject
+            {
+                {"Results", 0}
+            };
+
+        }
+
+        /// <summary>
+        /// Gets all the available hard drives and checks the free space available for all of them 
+        /// </summary>
+        /// <param name="monitorItem">The information for the monitor action. </param>
+        /// <param name="threshold">The threshold to check if an email needs to be send.</param>
+        /// <param name="gclCommunicationsService">The name of the service in the configuration, used for logging.</param>
+        /// <param name="configurationServiceName">The communications service from the GCL to actually send out the email.</param>
+        /// <returns></returns>
+        private async Task GetAllHardDrivesSpaceAsync(ServerMonitorModel monitorItem, int threshold, CommunicationsService gclCommunicationsService, string configurationServiceName)
+        {
+            foreach (var drive in allDrives)
+            {
+                if (!emailDrivesSent.ContainsKey(drive.Name))
+                {
+                    emailDrivesSent[drive.Name] = false;
+                }
+            }
+
+            foreach (var drive in allDrives)
+            {
+                //Calculate the percentage of free space availible and see if it matches with the given threshold.
+                double freeSpace = drive.TotalFreeSpace;
+                double fullSpace = drive.TotalSize;
+                double percentage = freeSpace / fullSpace * 100;
```

Comment:
Graag hier ook overal `var` gebruiken, zodat het consistent is met de rest van de code.